### PR TITLE
[Core] add norm_1 to amatrix interface

### PR DESCRIPTION
**Pull request by vryy**
add norm_1 to amatrix_interface.h

### CODE REVIEW

File: kratos/includes/amatrix_interface.h [hunk 1]
```diff
@@ -609,6 +609,19 @@ typename TExpression1Type::data_type inner_prod(
 
 }
 
+    template <typename TExpressionType, std::size_t TCategory>
+    typename TExpressionType::data_type norm_1(
+        AMatrix::MatrixExpression<TExpressionType, TCategory> const& TheExpression) {
+            using data_type = typename TExpressionType::data_type;
+            auto& the_expression = TheExpression;
```
Comment (pooyan-dadvand) [inline kratos/includes/amatrix_interface.h]: I don't know how it compiles like this....

It should be:
``` c++
auto& the_expression = TheExpression..expression();
```

**Review by pooyan-dadvand — CHANGES_REQUESTED** [review]
Would you please add a test for it?

File: kratos/includes/amatrix_interface.h (hunk 1 above)
Comment (vryy) [reply kratos/includes/amatrix_interface.h]: it also compiled with auto& the_expression = TheExpression; but I changed to your suggestion in the new commit

**Comment by vryy** [comment]
> Would you please add a test for it?

added in the new commit

**Comment by philbucher** [comment]
it does not compile

**Comment by vryy** [comment]
> it does not compile

@philbucher fixed in new commit 0cbf011c57bd1e8a05fc8074212d8bf437798cff

**Comment by vryy** [comment]
@pooyan-dadvand do you agree with my changes?

**Comment by pooyan-dadvand** [comment]
Is there any test for norm_1?

**Comment by vryy** [comment]
tests are implemented in tests/test_vector_interface.py

**Review by pooyan-dadvand — APPROVED** [review]
Thanks @vryy !

**Pull request merged**